src/OutputParser.elm

```diff
@@ -0,0 +1,240 @@
+module OutputParser exposing (parse)
+
+import Char
+import Html exposing (Html)
+import Json.Encode as JE
+import Parser exposing (..)
+import Set
+
+
+parse : String -> Result (List DeadEnd) JE.Value
+parse =
+    Parser.run parser
+
+
+
+-- STRING
+
+
+string : Parser JE.Value
+string =
+    succeed JE.string
+        |. symbol "\""
+        |= (getChompedString <| chompWhile ((/=) '"'))
+        |. symbol "\""
+
+
+
+-- BOOL
+
+
+bool : Parser JE.Value
+bool =
+    oneOf
+        [ keyword "True" |> map (\_ -> JE.bool True)
+        , keyword "False" |> map (\_ -> JE.bool False)
+        ]
+
+
+
+-- NUMBER
+
+
+float : Parser JE.Value
+float =
+    number
+        { int = Just toFloat
+        , hex = Nothing
+        , octal = Nothing
+        , binary = Nothing
+        , float = Just identity
+        }
+        |> map JE.float
+
+
+
+-- UNIT
+
+
+unit : Parser JE.Value
+unit =
+    symbol "()" |> map (\_ -> JE.null)
+
+
+
+-- LIST
+
+
+list : Parser JE.Value
+list =
+    sequence
+        { start = "["
+        , separator = ","
+        , end = "]"
+        , spaces = spaces
+        , item = value
+        , trailing = Forbidden
+        }
+        |> map (JE.list identity)
+
+
+
+-- OBJECT
+
+
+key : Parser String
+key =
+    variable
+        { start = Char.isLower
+        , inner = \c -> Char.isAlphaNum c || c == '_'
+        , reserved = Set.empty
+        }
+
+
+keyValue : Parser ( String, JE.Value )
+keyValue =
+    succeed Tuple.pair
+        |= key
+        |. spaces
+        |. symbol "="
+        |. spaces
+        |= value
+
+
+record : Parser JE.Value
+record =
+    sequence
+        { start = "{"
+        , separator = ","
+        , end = "}"
+        , spaces = spaces
+        , item = keyValue
+        , trailing = Forbidden
+        }
+        |> map JE.object
+
+
+
+-- INTERNALS
+
+
+internals : Parser JE.Value
+internals =
+    map JE.string <|
+        getChompedString <|
+            succeed ()
+                |. symbol "<"
+                |. chompWhile ((/=) '>')
+                |. symbol ">"
+
+
+
+-- TUPLES
+
+
+tuple : Parser JE.Value
+tuple =
+    sequence
+        { start = "("
+        , separator = ","
+        , end = ")"
+        , spaces = spaces
+        , item = value
+        , trailing = Forbidden
+        }
+        |> map (JE.list identity)
+
+
+
+-- UNION
+
+
+ctor : Parser JE.Value
+ctor =
+    map (JE.string << formatCtor) <|
+        variable
+            { start = Char.isUpper
+            , inner = \c -> Char.isAlphaNum c || c == '_'
```

Comment:
needs to be: ```            , inner = \c -> Char.isAlphaNum c || c == '_' || c == '.'``` so that e.g. `people = Dict.fromList []` will be parsed (as `ctorWithArgs`). Seems like `Dict`s, `Array`s and `Set`s are logged via their resp. `fromList` function